code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -406,13 +406,73 @@
 	name = "Ian"
 	real_name = "Ian"	//Intended to hold the name without altering it.
 	gender = MALE
-	desc = "It's a corgi."
+	desc = "It's the HoP's beloved corgi."
 	var/turns_since_scan = 0
 	var/obj/movement_target
 	response_help  = "pets"
 	response_disarm = "bops"
 	response_harm   = "kicks"
 	gold_core_spawnable = 0
+	var/age = 0
+	var/record_age = 1
+	var/memory_saved = 0
+	var/saved_head //path
+
+/mob/living/simple_animal/pet/dog/corgi/Ian/New()
+	Read_Memory()
+	if(age == 0)
+		var/mob/living/simple_animal/pet/dog/corgi/puppy/P = new /mob/living/simple_animal/pet/dog/corgi/puppy(loc)
+		P.name = "Ian"
+		P.real_name = "Ian"
+		P.gender = MALE
+		P.desc = "It's the HoP's beloved corgi puppy."
+		Write_Memory(0)
```

Comment:
Won't this prevent Puppy Ian's death or hat from being saved at the end of the round?